mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/parser/StringCombinators.kt

```diff
@@ -23,6 +25,36 @@ object StringCombinators {
   /** Parses the first digit of a [String], if it exists. */
   fun digit(): Parser<String, Int> = char().filter { it in '0'..'9' }.map { it - '0' }
 
+  /**
+   * Parses the first [Token] of a [String]
+   *
+   * @param delimiter the delimiter between each token
+   */
+  fun string(delimiter: Char = ' '): Parser<String, Token> = Parser {
+    if (it.isNotEmpty()) {
+      val splitString = it.trim().split(delimiter, limit = 2)
+      val result = splitString.first()
+      val remaining =
+          if (splitString.size == 1) {
+            ""
+          } else {
+            // drop only if we can drop something
+            splitString.drop(1).first()
+          }
+      sequenceOf(Parser.Result(remaining, result))
+    } else {
+      emptySequence()
+    }
+  }
+
+  /**
+   * Parses the first [Token] of a [String], if it's not empty and has the provided value.
+   *
+   * @param value the value that is searched.
+   */
+  fun string(value: Token, delimiter: Char = ' '): Parser<String, Token> =
```

Comment:
Same here.

Reply:
Fix in commit https://github.com/epfl-SDP/android/pull/280/commits/e5dd6733d5a1211cfed6e41538ea568c7e165cf0